primefaces/src/main/java/org/primefaces/component/splitbutton/SplitButtonRenderer.java

```diff
@@ -98,10 +98,8 @@ protected void encodeMarkup(FacesContext context, SplitButton button) throws IOE
         }
 
         encodeDefaultButton(context, button, buttonId);
-        if (button.getElementsCount() > 0) {
```

Comment:
By removing this check it will now always render the menu?  Or do we need to add more short circuits inside the encodeMenuIcon and encodeMenu methods?

Reply:
Yes, it will always render the icon (drop down button) and the menu. I don't need the menu, and probably the code is now also toggling an empty div, but this was the least intrusive fix I could come up with.

Reply:
-1 If you look at the history it revert 3 tickets, also if no element are present in splitButton it should be displayed as a simple button

Reply:
If it reverts 3 tickets we need integration tests for those 3 tickets and this new one.

Reply:
It currently does not display as a simple button properly when there are no items (but that's a theme issue).

Reply:
So https://github.com/primefaces/primefaces/commit/07412082bb0218784032a022f5cf49fec0cc8050 solved 3 issues, but I'm pretty sure that removing the `if` won't regress https://github.com/primefaces/primefaces/issues/4104, https://github.com/primefaces/primefaces/issues/4105 and https://github.com/primefaces/primefaces/issues/4105

Reply:
Ok, I see that very same line had replaced what I did something in the past (see #2690) so there must be a reason for it. Arf...and also https://github.com/primefaces/primefaces/commit/07412082bb0218784032a022f5cf49fec0cc8050 introduced already a regression on #2690

Reply:
Alright. I can add a boolean attribute for it. But there also will need to be a change in the theme in that case. I'm not sure it only is the border radius, but at least the radius is 0 at the right side of the button (did not check RTL).

Reply:
Yeah, this shows the dropdown currently:
```java
<p:splitButton id="noRenderedChildren" value="No rendered children">
    <p:menuitem value="Rendered false" rendered="false"/>
</p:splitButton>
```
Should I create a separate ticket for this? I will fix this in this PR though.

Reply:
If you have plans to fix also the css part (or that's something up to PrimeTek?) maybe in another issue then, otherwise I don't mind if you fix it in this PR

Reply:
CSS must be fixed by PrimeTek (it's part of the theme), so that will need an issue for sure.

Reply:
I haven't looked but now that this exists can we start fixing the CSS ourselves?  https://github.com/primefaces/primefaces-sass-theme

Reply:
Ha, yeah, could be. I could open a ticket there and see if I can provide a PR. And we need a readme.

Reply:
Yep and I was gonna make a pom.xml that uses Dart SASS and minifies the CSS.